<!DOCTYPE HTML PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml"><head>
  <meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
  <title>MIM Pathvisio</title>
  <link href="mim_pathvisio.css" rel="stylesheet" type="text/css">
</head><body>

<div id="overallbox">
<a name="global_nav"></a>
<div id="logo"><a href="http://discover.nci.nih.gov/mim/mim_pathvisio.html">MIM Pathvisio</a></div>

<div id="nav"> 
<li><a href="#introduction">Introduction</a></li>
<li><a href="#usage">Usage</a></li>
<li><a href="#usage_tips">Usage Tips</a></li>
<li><a href="#downloads">Download</a></li>
<li><a href="#file_formats">File Formats</a></li>
<li><a href="#authors">Authors</a></li>
</div>

<div class="box">
<h1><a name="introduction"></a>What is the MIM Pathvisio plugin?</h1>

<p>
The MIM (Molecular Interaction Map) plugin for <a href="http://www.pathvisio.org/">Pathvisio</a> allows users to produce diagrams of bioregulatory systems according to an implementation of notation first developed by Kurt Kohn in 1999. The plugin has two main features: 1) it provides users with the glyphs necessary to draw MIM diagrams and 2) it allows users to import and export to the MIM Markup Language (MIMML) format, a format developed to exchange MIM diagrams between MIM compatible tools and used to validate MIM diagrams. Diagrams produced using the MIM Pathvisio plugin can be exported into several image formats including PDF, PNG, SVG, and TIFF.</p>

<h1>What is the MIM notation?</h1>
<p>The MIM notation is a graphical notation for biological systems for the representation of networks containing multi-protein complexes, protein modifications, and enzymes that are substrates of other enzymes. This graphical representation makes it possible to view all of the many interactions in which a given molecule may be involved, and it can portray competing interactions, which are common in bioregulatory networks. One aim of the MIM notation is to standardize diagrams of biological systems with a fixed set of glyphs with specific semantics. The specific semantics of the MIM notation can be translated into other standardized formats.</p> 

<h1><a name="example"></a>What do the MIM diagrams look like?</h1>
<p>Here are two sample images: the <a href="img/chk2.png">activation of Chk2 in response to double strand breaks</a> and an <a href="img/example.png">example diagram</a> (shown below) showing all the various elements of the MIM notation, as implemented.<img class="diagram" alt="example diagram" src="img/example.png"></p> 

<h1>Who are the intended users of the MIM Pathvisio plugin?</h1>
<p>Researchers interested in using the MIM notation to diagram their biological systems. This would include researchers that want their diagrams to have a well-defined semantic attached to the elements in their diagrams. Because the MIM notation is a graphical notation for bioregulatory networks in general, the type of experiment or study is irrelevant.
</p>

<h1>What are the main features of Pathvisio and the MIM Pathvisio plugin?</h1>

<p><h2>Pathvisio</h2>
	<ul>
		<li>Publishable-quality graphics in resolution-independent formats, such as PDFs</li>
		<li>Extensible software architecture</li>
		<li>Capable of <a href="http://www.pathvisio.org/wiki/Cytoscape_plugin">converting diagrams</a> to <a href="http://www.cytoscape.org/">Cytoscape</a>  networks</li>
		<li>Ability to map microarray data to diagrams</li>
		<li>Ability diagram nodes to external database references</li>
		<li>Ability to annotate diagrams with comments</li>
		<li>Free and open source</li>
	</ul>
	<h2>MIM Pathvisio plugin</h2>
	<ul>
		<li>Provides the ability use MIM glyphs for diagramming bioregulatory systems in Pathvisio</li>
		<li>Imports and exports in the MIM exchange format, MIMML</li>
		<li>Free and open source</li>
	</ul>
</p>

<br>
<a href="#global_nav"><img class="topLink" alt="up_arrow.png" src="img/up_arrow.png"></a>
</div>
<div class="box">

<h1><a name="usage"></a>How do I install the MIM Pathvisio plugin?</h1>

<p>The MIM Pathvisio plugin requires the installation of Pathvisio, freely available diagram editor for biological systems, found <a href="http://www.pathvisio.org/">here</a> for Linux, Windows, or MacOS. The MIM Pathvisio plugin uses two Java .jar files that can be downloaded <a href="#downloads">here.</a> These .jar files must be placed in the plugins folder of Pathvisio:</p>

<p>In <strong>Windows XP,</strong> make sure that you have downloaded the .jar files to the following location. The word <strong>USER</strong> should be the name of the user for whom the plugin will be installed. 
</p><pre>C:\Documents and Settings\USER\.Pathvisio\plugins</pre>

In <strong>Windows Vista/7,</strong> make sure that you have downloaded the .jar files to the following location. The word <strong>USER</strong> should be the name of the user for whom the plugin will be installed. 
</p><pre>C:\Users\USER\.Pathvisio\plugins</pre>

In <strong>OS X/Unix/Linux,</strong> make sure that you have downloaded the .jar files to the following location. The  <strong>~</strong> symbol represents the home directory of the user for whom the plugin will be installed. 
<pre>~/.Pathvisio/plugins</pre>

Any errors in loading the .jar files will be logged in the <strong>Pathvisio.log</strong> file located in the <strong>.Pathvisio</strong> folder. 

The MIM Pathvisio plugin can be uninstalled by deleting the MIM plugin .jar files.

<h1><a name="usage"></a>How is the MIM Pathvisio plugin used?</h1>

<p>
Once the MIM Pathvisio plugin has been installed, the MIM Pathvisio plugin will be loaded whenever Pathvisio is run. The user interface of Pathvisio should have two additions, which are shown below. A new toolbar should be visible near the top of the Pathvisio editor. From left to right, the glyphs are simple physical entity, simple physical entity with features, conceptual entity, source/sink, modifier, and restricted copy. Additionally, the MIM Pathvisio plugin adds a series of MIM-specific line types available to a user when a line is selected. The MIM-specific line types are prefixed with <strong>mim-</strong> and include the following: 

<ul>
	<li>mim-necessary-stimulation</li>
	<li>mim-binding: Corresponds to non-covalent irreversible binding</li>
	<li>mim-conversion: Corresponds to stochiometric conversion</li>
	<li>mim-stimulation</li>
	<li>mim-modification: Corresponds to covalent modification</li>
	<li>mim-catalysis</li>
	<li>mim-inhibition</li>
	<li>mim-cleavage</li> 
	<li>mim-covalent-bond: Corresponds to covalent irreversible binding</li>
	<li>mim-branching-left</li>
	<li>mim-branching-right</li>
	<li>mim-transcription-translation: Corresponds to template reaction</li>
	<li>mim-gap: <strong>Note: currently, not supported during validation</strong></li>
	<li>mim-absolute-inhibition</li>
	<li>mim-production-wo-loss: Corresponds to production without loss</li>
	<li>mim-first-feature</li>
	<li>mim-next-feature</li>
	<li>mim-state-combination</li>
</ul>

<img class="diagram" alt="plugin interface" src="img/mim_plugin.png">
</p>

<p>Explanation of the MIM Pathvisio plugin input files is provided in the <a href="#file_formats">file formats</a>
section. Example diagrams are provided as <a href="#downloads">MIMML</a> files. 

</p>

<br>
<a href="#global_nav"><img class="topLink" alt="up_arrow.png" src="img/up_arrow.png"></a>
</div>
<div class="box">

<h1><a name="usage_tips"></a>How do I ...?</h1> 
<p>Pathvisio provides <a href="http://www.pathvisio.org/wiki/PathVisioHelp">documentation</a> for many of its capabilities. This document describes the usage of the features specific to the MIM Pathvisio plugin.</p>

<h1>How do I draw a MIM interaction?</h1>
<p>MIM interactions are lines in Pathvisio with the correct start and end line types. Both the start and end line types must correspond to a MIM interaction. For example, a catalysis interaction is a line with <strong>mim-catalysis</strong> and a <strong>Line</strong> line types; either line type may be chosen as the start or end type. In the case of non-covalent reversible binding, both ends of the interactions must use the mim-binding line types.</p>

<h1>How do I know which MIM glyph I should use?</h1> 
<p>Below are example of possible uses for MIM entity glyphs; a detailed explanation of the MIM notation and glyph usage can be found in the [[MIM_SPECIFICATION]]. The MIM glyphs 
	<h2>Entity Glyphs</h2>
	<ul>
		<li>Simple physical entity: Protein or DNA</li>
		<li>Entity feature: A region of a protein or DNA</li>
		<li>Conceptual entity: A black-box pathway or external stimuli</li>
		<li>Source/sink: An entity pool that acts as the input to an interaction (source) or a pool of the products of a degradation (source)</li>
		<li>Modifier: A small molecule such as a phosphate or ubiquitin molecule</li>
		<li>Restricted copy: Used to represent homodimers and homo-multimers</li>	
	</ul>
</p>

<h1>How do I indicate that my simple physical entity is a protein, DNA, etc?</h1>
<p>The MIM Pathvisio plugin creates an additional entry in the Properties panel whenever a entity glyph is selected in Pathvisio called <strong>Entity Controlled Vocabulary</strong>. The following values may be used and are taken from the <a href="http://www.biopax.org">BioPAX</a> specification: 
	<h2>Entity Controlled Vocabulary</h2>
	<ul>
		<li>Complex</li>
		<li>RNA</li>
		<li>RNARegion</li>
		<li>DNA</li>
		<li>DNARegion</li>
		<li>Protein</li>
		<li>SmallMolecule</li>
	</ul>
</p>

<h1>How do I draw proteins with domains?</h1> 
<p>A protein with domains is presented as a simple physical entity connected to an interaction with a first-feature arrowhead pointing towards the protein that will possess domains. On the opposite end of the interaction an entity feature should be attached via the Line line type; the entity feature will represent the protein domain. Entity features are created first producing a simple physical entity and then change its <strong>Type</strong> in the Properties panel. Subsequent domains are connected by interactions with next-feature arrowheads on the end connected to the next feature and a Line line type on the other end. The <a href="#example">example</a> diagram shows an entity with two features B and C that could be used to graphically represent a protein with two domains.</p>

<h1>How do I indicate the location of a domain within a protein?</h1> 
<p>Protein domains are represented as features of a simple physical entity. Entity A in the <a href="#example">example</a> diagram has two entity features B and C. Entity feature C is a region on simple physical entity A that extends from position 10 to position 20. The labels of entity features should have one of the following formats: X, Y:X, Y:X:Z, or X:Z; where X is a text label, Y is an integer start location value, and Z is an integer end location value.</p>

<h1>How do I represent acetylations, phosphorylations, methylations, etc?</h1> 
<p>There are several pre-defined values for covalent modifiers of simple physical entities. These values should be used to increase the readability of diagrams. 
	<h2>Common values for covalent modification labels</h2>
	<ul>
		<li>Ac: acetylation</li>
		<li>G: glycosylation</li>
		<li>OH: hydroxylation</li>
		<li>Me: methylation</li>
		<li>My: myristoylation</li>
		<li>Pa: palmytoylation</li>
		<li>P: phosphorylation</li>
		<li>Pr: prenylation</li>
		<li>H: protonation</li>
		<li>Ub: ubquitination</li>
	</ul>
</p>

<h1>How do I indicate the location of covalent modifications, such as phosphorylations?</h1> 
<p>Covalent modifications are specified as a covalent modification interaction connecting a modifier and a simple physical entity or entity feature. Entity A in the <a href="#example">example</a> diagram is modified at two locations: positions 1 and 2. These are indicated as part of the labels of the modifiers. The labels of modifiers should have one of the following formats: X, X:YZ, or X:Z; where X is a text label, Y is a capitalized single letter amino acid abbreviation, and Z is an integer location value.</p>

<h1>How do I draw homodimers?</h1> 
<p>It is encouraged that entities be represented only once on a diagram. A homodimer is represented as a reversible binding interaction between a protein and a restricted copy entity. A dimer of H is shown on the <a href="#example">example</a> diagram.</p>

<h1>How do I draw a gene transcribed to mRNA then translated to a protein?</h1>
<p>The transcription/translation arrowhead is used to describe both processes. Alternatively, to save space the production without loss arrowhead may be used to collapse this representation into a single interaction. This is shown in the <a href="#example">example</a> diagram as entity G being produced from K without the loss of any K molecules. <strong>Note: This does not carry the same semantics as the production of K from the template J.</strong></p>

<h1>How do I draw branched interactions?</h1> 
<p>Examples of the proper usage of branched interactions are shown in the <a href="#example">example</a> diagram. For instance, entity G is converted into entities N and the degradation products represented by the sink glyph. Graphically, the source of a branched interaction is represented as the direction of the branch. Currently, this semantic is not reproduced in MIMML files. Users may indicate the start of a branched interaction using the <strong>Interaction Start</strong> property in the Properties panel when an interaction is selected in Pathvisio.</p>

<h1>How do I incorporate the the MIM figures into my publication?</h1>
<p>The listing below shows the most common scenarios for inclusion of a
figure into a publication; methods will vary based on the intended
publication type:
</p>

<ol>
<li><strong>Inclusion of PDF into Word document:</strong> PDF images can be selected and copied using the Adobe Reader snapshot tool <img alt="snapshot_tool.png" src="img/snapshot_tool.png">. These images can then be pasted into a Word document. 

</li><li><strong>Inclusion of PDF/EPS into LaTeX document:</strong> Images included into DVI file using <strong>latex</strong> must be in the EPS format. PDF documents created by <strong>pdflatex</strong> cannot use files in the EPS format and requires the PDF format. The following is basic LaTeX code for including an image. <strong>FILENAME</strong> must be changed to the appropriate filename. 
<pre>\documentclass{article}

\usepackage{graphicx}

\begin{document}
\begin{figure}
\includegraphics{FILENAME}
\end{figure}
\end{document}
</pre> 
</li>

<li><strong>Conversion of PDF/EPS file into other formats using Ghostscript</strong>:
Ghostscript can be used to convert PDF and EPS files into a variety of
raster formats such as JPG, BMP, PNG, etc. The following commandline
converts an EPS image into a JPG image; the resulting image has a
resolution of 300 DPI and pixel dimension of 1050x1050 pixels. Further
information on the usage of Ghostscript can be found <a href="http://www.cs.wisc.edu/%7Eghost/doc/AFPL/8.00/Use.htm">here.</a> <strong>FILENAME</strong> must be changed to the appropriate filename. 
<pre>gs -q -dNOPAUSE -dSAFER -dBATCH -r300 -g1050x1050 -sOutputFile=FILENAME.jpg -sDEVICE=jpeg FILENAME.eps
</pre> 
</li>
</ol>
<br>
<a href="#global_nav"><img class="topLink" alt="up_arrow.png" src="img/up_arrow.png"></a>
</div>

<div class="box">
<h1><a name="file_formats"></a>What file types are used by the MIM Pathvisio plugin?</h1>

<p>Users of the MIM Pathvisio plugin can import diagrams in any format that is supported Pathvisio, which includes Pathvisio's native formats GPML (.gpml) or GenMAPP (.mapp), and an additional <a href="http://pathvisio.com/wiki/BiopaxPluginHelp">Biopax plugin</a> supports the <a href="http://www.biopax.org">BioPAX format</a>. The MIM Pathvisio Plugin add the capability to import and export the MIMML format which is based on the GPML format with modifications to support the validation of MIM diagrams represented by the MIMML files. The MIMML format is capable of representing the subset of the glyphs available in Pathvisio that exist in the MIM notation.</p>

<p>Validation of MIMML files is done using a <a href="http://www.schematron.com/">Schematron</a>-formatted rule set. The rule set can be used in any environment with an <a href="http://en.wikipedia.org/wiki/Xslt">XSLT</a> processor. The rule set is processed into an XSLT stylesheet that is then run against a MIMML file to generate a validation report. The sample command below uses the <a href="http://xalan.apache.org/index.html">Xalan XSLT processor</a>

<pre>Xalan mimml_validation.sch ./schematron/iso_svrl_for_xslt1.xsl > mimml_validation.xsl; Xalan example.mimml mimml_validation.xsl</pre> 
</p>
<br>
<a href="#global_nav"><img class="topLink" alt="up_arrow.png" src="img/up_arrow.png"></a>
</div>

<div class="box">

<h1><a name="downloads"></a>Where can I download the MIM Pathvisio plugin and how can I get the source code for my own project?</h1>

<ul>
  <li><a href="https://ncisvn.nci.nih.gov/svn/mim/trunk/mim_plugin/MIMPathvisio.jar">Download the MIM Pathvisio .jar file to add MIM-specific glyphs to Pathvisio</a></li>
  <li><a href="https://ncisvn.nci.nih.gov/svn/mim/trunk/mim_gpml/MIMGPML.jar">Download the MIM Pathvisio .jar file to add MIMML import and export capability to Pathvisio</a>
  <li><a href="https://ncisvn.nci.nih.gov/svn/mim/trunk/">An online subversion repository browser for the MIM source code</a></li>  
  <li><a href="https://ncisvn.nci.nih.gov/svn/mim/trunk/">Download the MIMML Schematron rule set</a>[[NOT AVAILABLE]]</li>
  <li>Download MIMML files: <a href="files/chk2.mimml">Chk2</a> and <a href="files/example.mimml">MIM example</a> diagrams</li>
  <li><a href="mailto:?Subject=Interesting%20Diagram%20Editor:%20the MIM Pathvisio plugin&amp;Body=Hi%2C%0D%0A%0D%0AI%20think%20you%27ll%20find%20the%20the MIM Pathvisio plugin%20interesting,%0D%0Ahttp://discover.nci.nih.gov/mim/mim_pathvisio.html%20It%20creates%20publishable-quality%20diagrams%20of%20biological%20systems%20using%20the%20MIM%20graphical%20notation.%20Check%20it%20out%21%0D%0A%0D%0ARegards%2C%0D%0A">Tell a friend</a></li>
  <li><a href="http://www.pathvisio.org/">Learn about Pathvisio and Wikipathways</a></li>
  <li><a href="http://www.pathvisio.org/">Learn about Schematron</a></li>
  <li><a href="mailto:augustin@mail.nih.gov?subject=MIM%20Pathvisio%20Plugin%20Bug%20Report">Report a bug</a></li>
</ul>

<a href="#global_nav"><img class="topLink" alt="up arrow" src="img/up_arrow.png"></a>
</div>

<div class="box">
<h1><a name="authors"></a>Authors</h1>
<p>
Augustin Luna, BS<br>
2005-2007: Postbaccalaureate Intramural Research Training Award Fellow<br>
Clinical Brain Disorders Branch (CBDB)/Genes, Cognition, and Psychosis
Program (GCAP), National Institute of Mental Health (NIMH), National
Institutes of Health (NIH) <br><br>
2007-Current: Pre-Doctoral Intramural Research Training Award Fellow<br>
Center for Cancer Research (CCR), Laboratory of Molecular Pharmacology (LMP), National Cancer Institute (NCI), National
Institutes of Health (NIH) <br><br>
BS, Biomedical Engineering, Georgia Institute of Technology <br>
PhD candidate, Bioinformatics, Boston University<br>
</p>
<p>Margot Sunshine</p>
<p><a href="http://www3.cancer.gov/intra/lmp/kwkbio.htm">Kurt Kohn</a></p>
<p><a href="http://www3.cancer.gov/intra/lmp/MABIO.HTM">Mirit Aladjem</a></p>
<br>
<a href="#global_nav"><img class="topLink" alt="up_arrow.png" src="img/up_arrow.png"></a>
</p></div>
</div>

</body></html>